### PR TITLE
Fix/#175 do not save secrets in entities

### PR DESCRIPTION
**Pull request by jrobinAV**
I implemented a fix for taipy-core #175.

From now, the Config._applied_config object still does contain the compiled config, but the environment values are not interpreted. For instance, the config field contains values like 'ENV[NAME]' (before, the value was interpreted and replaced by the value of the corresponding environment variable).
I then needed to ensure that every time a config field is accessed, it is interpreted on the fly through the method: _tpl._replace_templates. I created getters/setters properties for that purpose.

Here is the expected behavior of the usage of a config object:
```
def test_data_node_with_env_variable_value():
    with mock.patch.dict(os.environ, {"FOO": "pickle", "BAR": "baz"}):
        Config.configure_data_node("data_node", storage_type="ENV[FOO]", prop="ENV[BAR]")
        assert Config.data_nodes["data_node"].prop == "baz"
        assert Config.data_nodes["data_node"].properties["prop"] == "baz"
        assert Config.data_nodes["data_node"]._properties["prop"] == "ENV[BAR]"
        assert Config.data_nodes["data_node"].storage_type == "pickle"
        assert Config.data_nodes["data_node"]._storage_type == "ENV[FOO]"
```

First consequence: When exporting the applied config, we do not export the env variable values, but the templates. And so secrets are not serialized.

Second consequence: When creating an entity, the properties of the config are copied before they got interpreted and replaced. When serializing the entity through a repository, we do not serialize interpreted values. And so secrets are not serialized.
I then needed to ensure that every time an entity field is accessed, it is interpreted on the fly through the method: _tpl._replace_templates. I created getters/setters properties for that purpose.

Here is the behavior of the usage of an entity object:
```
def test_data_node_with_env_variable_value_not_serialized(self):
    with mock.patch.dict(os.environ, {"FOO": "bar"}):
        assert self.data_node._properties["prop"] == "ENV[FOO]"
        assert self.data_node.properties["prop"] == "bar"
        assert self.data_node.prop == "bar"
```

### CODE REVIEW

**Review by khaitruong922 — DISMISSED** [review]

**Comment by jrobinAV** [comment]
After better thinking, I believe the following is not great:
data_node.properties["prop"] != data_node.prop

File: taipy/core/config/_config_template_handler.py [hunk 1]
```diff
@@ -30,6 +31,8 @@ def _replace_templates(cls, template, type=str, required=True, default=None):
             return [cls._replace_template(item, type, required, default) for item in template]
         if isinstance(template, dict):
             return {str(k): cls._replace_template(v, type, required, default) for k, v in template.items()}
+        if isinstance(template, _Properties):
+            return {str(k): cls._replace_template(v, type, required, default) for k, v in template.items()}
```
Comment (florian-vuillemot) [inline taipy/core/config/_config_template_handler.py:35]:
```suggestion
        if isinstance(template, dict) or isinstance(template, _Properties):
            return {str(k): cls._replace_template(v, type, required, default) for k, v in template.items()}
```

File: taipy/core/config/data_node_config.py [hunk 2]
```diff
@@ -51,49 +51,72 @@ class DataNodeConfig:
...
+    @storage_type.setter  # type: ignore
+    def storage_type(self, val):
+        self._storage_type = val
+
+    @property
+    def properties(self):
+        res = {}
+        for k, v in self._properties.items():
+            res[k] = _tpl._replace_templates(v)
+        return res
```
Comment (florian-vuillemot) [inline taipy/core/config/data_node_config.py]:
```suggestion
        return {k: _tpl._replace_templates(v) for k, v in self._properties.items()}
```

File: taipy/core/config/task_config.py [hunk 3]
```diff
@@ -58,10 +58,21 @@ def __init__(
...
     def __copy__(self):
-        return TaskConfig(self.id, copy(self._inputs), self.function, copy(self._outputs), **copy(self.properties))
+        return TaskConfig(self.id, copy(self._inputs), self.function, copy(self._outputs), **copy(self._properties))
+
+    @property
+    def properties(self):
+        res = {}
+        for k, v in self._properties.items():
+            res[k] = _tpl._replace_templates(v)
+        return res
```
Comment (florian-vuillemot) [inline taipy/core/config/task_config.py]:
```suggestion
        return {k: _tpl._replace_templates(v) for k, v in self._properties.items()}
```

File: tests/core/data/test_data_manager.py [hunk 4]
```diff
@@ -36,6 +36,7 @@ def test_create_data_node_and_modify_properties_does_not_modify_config(self):
         assert dn_config.properties.get("foo") == "bar"
         assert dn_config.properties.get("baz") is None
 
+        # dn._properties["baz"] = "qux"
```
Comment (florian-vuillemot) [inline tests/core/data/test_data_manager.py]: ?

**Review by florian-vuillemot — DISMISSED** [review]

File: taipy/core/config/global_app_config.py [hunk 5]
```diff
@@ -47,51 +47,83 @@ def __init__(
...
+    @clean_entities_enabled.setter  # type: ignore
+    def clean_entities_enabled(self, val):
+        self._clean_entities_enabled = val
+
+    @property
+    def properties(self):
+        res = {}
+        for k, v in self._properties.items():
+            res[k] = _tpl._replace_templates(v)
+        return res
```
Comment (jrobinAV) [inline taipy/core/config/global_app_config.py]:
```suggestion
return {k: _tpl._replace_templates(v) for k, v in self._properties.items()}
```

File: taipy/core/config/pipeline_config.py [hunk 6]
```diff
@@ -31,17 +31,28 @@ class PipelineConfig:
...
     def __copy__(self):
-        return PipelineConfig(self.id, copy(self._tasks), **copy(self.properties))
+        return PipelineConfig(self.id, copy(self._tasks), **copy(self._properties))
+
+    @property
+    def properties(self):
+        res = {}
+        for k, v in self._properties.items():
+            res[k] = _tpl._replace_templates(v)
+        return res
```
Comment (jrobinAV) [inline taipy/core/config/pipeline_config.py]:
```suggestion
return {k: _tpl._replace_templates(v) for k, v in self._properties.items()}
```

File: taipy/core/config/scenario_config.py [hunk 7]
```diff
@@ -59,11 +59,22 @@ def __init__(
...
         comp = None if self.comparators is None else self.comparators
-        return ScenarioConfig(self.id, copy(self._pipelines), self.frequency, copy(comp), **copy(self.properties))
+        return ScenarioConfig(self.id, copy(self._pipelines), self.frequency, copy(comp), **copy(self._properties))
+
+    @property
+    def properties(self):
+        res = {}
+        for k, v in self._properties.items():
+            res[k] = _tpl._replace_templates(v)
+        return res
```
Comment (jrobinAV) [inline taipy/core/config/scenario_config.py]:
```suggestion
return {k: _tpl._replace_templates(v) for k, v in self._properties.items()}
```

File: tests/core/data/test_data_manager.py (hunk 4 above)
Comment (jrobinAV) [inline tests/core/data/test_data_manager.py]:
```suggestion
```

**Comment by jrobinAV** [comment]
> After better thinking, I believe the following is not great: data_node.properties["prop"] != data_node.prop

This is fixed!

**Pull request merged**